codebasin/report.py

```diff
@@ -244,3 +249,72 @@ def clustering(output_name, setmap):
         fig.savefig(fp)
 
     return "\n".join(lines)
+
+
+def find_duplicates(codebase: CodeBase) -> list[set[Path]]:
+    """
+    Search for duplicate files in the code base.
+
+    Returns
+    -------
+    list[set[Path]]
+        A list of all sets of Paths with identical contents.
+    """
+    # Search for possible matches using a hash, ignoring symlinks.
+    possible_matches = defaultdict(set)
+    for path in codebase:
+        path = Path(path)
+        if path.is_symlink():
+            continue
+        with open(path, "rb") as f:
+            digest = hashlib.file_digest(f, "sha512").hexdigest()
+        possible_matches[digest].add(path)
+
+    # Confirm equality for files with the same hash.
+    confirmed_matches = []
+    for digest, paths in possible_matches.items():
```

Comment:
Maybe rename to `path_set`?

Reply:
Fixed in ae8977f3420ab8ad412a9c7f809ed32c4c61f4a7.